### PR TITLE
added for MDM UEM support

### PR DESCRIPTION
**Pull request by michaelcretzman**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md [hunk 1]
```diff
@@ -12,19 +12,110 @@ You can run shell scripts in a CD stage using the **Shell Script** step.
...
-* [Define Your Kubernetes Target Infrastructure](../../cd-infrastructure/kubernetes-infra/define-your-kubernetes-target-infrastructure.md): You can run a Shell Script step in a CD stage without specifying specs or artifacts, but you do need to set up the Infrastructure Definition.
+See [Shell Script Step Reference](../../cd-technical-reference/cd-gen-ref-category/shell-script-step.md).
 
-## Limitations
+## Use cases
 
-See [Shell Script Step Reference](../../cd-technical-reference/cd-gen-ref-category/shell-script-step.md).
+Typically, the primary deployment operations are handled by the default Harness deployment steps, such as the [Kubernetes Rollout Step](../../cd-technical-reference/cd-k8s-ref/kubernetes-rollout-step.md).
+
+The Shell Script step can be used for secondary options. There are several secondary scripts that DevOps teams commonly run in a Kubernetes container as part of a Continuous Deployment (CD) pipeline. These scripts can be used to perform various tasks such as configuration, data migration, database schema updates, and more.
```
Comment (KimberlyFields) [inline docs/continuous-delivery/cd-execution/cd-general-steps/using-shell-scripts.md:28]: Continuous Deployment should not be init capped if you're referring to CD as a strategy. Only init cap if referring to the Harness module.

File: docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md [hunk 2]
```diff
@@ -10,9 +10,99 @@ Harness orchestrates the container in your cluster as part of your Harness deplo
 
 ## Use cases
 
-A common use case of the Container step is running tests on your cluster.
+Typically, the primary deployment operations are handled by the default Harness deployment steps, such as the [Kubernetes Rollout Step](../../cd-technical-reference/cd-k8s-ref/kubernetes-rollout-step.md).
 
-Another example is implementing Mobile Device Management (MDM)/User Experience Management (UEM). You can leverage the Container step and call the MDM/UEM APIs from Harness.
+The Container step can be used for secondary options. There are several secondary scripts that DevOps teams commonly run in a Kubernetes container as part of a Continuous Deployment (CD) pipeline. These scripts can be used to perform various tasks such as configuration, data migration, database schema updates, and more.
```
Comment (KimberlyFields) [inline docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md:15]: Continuous Deployment should not be init capped if you're referring to CD as a strategy. Only init cap if referring to the Harness module.

**Review by KimberlyFields — APPROVED** [review]
Made a few suggestions, but approving.

File: docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md (hunk 2 above)
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md:15]: done

File: docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md [hunk 3]
```diff
@@ -10,9 +10,99 @@ Harness orchestrates the container in your cluster as part of your Harness deplo
...
+
+Some common secondary scripts that are run in a Kubernetes container as part of CD are:
+
+- Database migrations: These scripts are used to update the database schema and apply any necessary data migrations.
+- Configuration updates: These scripts are used to update the configuration of the application or service being deployed.
+- Health checks: These scripts are used to perform health checks on the application or service being deployed, ensuring that it is running correctly and responding to requests.
+- Load testing: These scripts are used to perform load testing on the application or service, ensuring that it can handle the expected traffic and load.
+- Monitoring setup: These scripts are used to set up monitoring and logging for the application or service, so that any issues can be quickly detected and addressed.
+- Smoke tests: These scripts are used to perform simple tests to confirm that the application or service is running correctly after deployment.
+- Cleanup: These scripts are used to clean up any resources or files that were created during the deployment process.
```
[the remark this answers is not in the document]
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md:28]: done

File: docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md [hunk 4]
```diff
@@ -10,9 +10,99 @@ Harness orchestrates the container in your cluster as part of your Harness deplo
...
+
+These secondary scripts are usually run as part of the CD pipeline, either as part of the build process or as separate jobs. They can be written in a variety of scripting languages. In many cases, these scripts are run in containers within the Kubernetes cluster, so that the necessary dependencies and tools are available.
+
+</details>
+
+### Mobile Device Management (MDM)/User Experience Management (UEM)
+
+Harness supports MDM/UEM through interaction with their APIs.
```
[the remark this answers is not in the document]
Comment (michaelcretzman) [reply docs/continuous-delivery/cd-execution/cd-general-steps/container-step.md:36]: fixed.

**Pull request merged**